src/ansys/aedt/core/application/aedt_file_management.py

```diff
@@ -47,125 +53,128 @@ def read_info_fromcsv(projdir, name):
 
     Parameters
     ----------
-    origfile : str
-        Full path to the project file, which has an ``.aedt``  extension.
-    solid_list : list
-        List of names for the solid to make visible. All other solides are hidden.
+    input_file : str or :class:`pathlib.Path`
+        Full path to the project file, which has an ``.aedt`` extension.
+    assignment : list
+        List of names for the solid to make visible. All other solids are hidden.
 
     Returns
     -------
     bool
         ``True`` when successful, ``False`` when failed.
     """
-    path, filename = os.path.split(origfile)
-    newfile = os.path.join(path, "aedttmp.tmp")
-
-    if not os.path.isfile(origfile + ".lock"):  # check if the project is closed
-        with open(origfile, "rb") as f, open(newfile, "wb") as n:
-            # Reading file content
-            content = f.read()
-
-            # Searching file content for pattern
-            pattern = re.compile(
-                r"(\$begin 'EditorWindow'\n.+)(Drawings\[.+\])(.+\n\s*\$end 'EditorWindow')", re.UNICODE
-            )
-            # Replacing string
-            # fmt: off
-            view_str = u"Drawings[" + str(len(solid_list)) + u": " + str(solid_list).strip("[")
-            s = pattern.sub(r"\1" + view_str + r"\3", content)
-            # fmt: on
-            # writing file content
-            n.write(str(s))
-
-        # renaming files and deleting temp
-
-        os.remove(origfile)
-        os.rename(newfile, origfile)
-
-    else:  # project is locked
-        print("change_objects_visibility: Project %s is still locked." % origfile)
-
-
-@pyaedt_function_handler()
-def change_model_orientation(origfile, bottom_dir):
+    path = Path(input_file).parent
+    newfile = path / "aedttmp.tmp"
+
+    if not (Path(input_file).with_suffix(".lock")).is_file():  # check if the project is closed
+        try:
+            # Using text mode with explicit encoding instead of binary mode.
+            with open(str(input_file), "rb") as f, open(str(newfile), "wb") as n:
+                # Reading file content
+                content = f.read()
+                # Searching file content for pattern
+                search_str = r"(\$begin 'EditorWindow'\n.+)(Drawings\[.+\])(.+\n\s*\$end 'EditorWindow')"
+                # Replacing string
+                view_str = "Drawings[" + str(len(assignment)) + ": " + str(assignment).strip("[")
+                sub_str = r"\1" + view_str + r"\3"
+                s = re.sub(search_str.encode("utf-8"), sub_str.encode("utf-8"), content)
+                # Writing file content
+                n.write(s)
+            # Renaming files and deleting temporary file
+            Path(input_file).unlink()
+            newfile.rename(input_file)
+            return True
+        except Exception:  # pragma: no cover
+            # Cleanup temporary file if exists.
+            if newfile.exists():
+                newfile.unlink()
```

Comment:
In Python 3.8+ the following should be the most pythonic way. 
```suggestion
            newfile.unlink(missing_ok=True)
```